x-pack/plugins/triggers_actions_ui/public/application/components/builtin_alert_types/threshold/lib/time_buckets/calc_es_interval.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import dateMath from '@elastic/datemath';
+
+import { parseEsInterval } from '../../../../../../../../../../src/plugins/data/public';
+
+const unitsDesc = dateMath.unitsDesc;
+const largeMax = unitsDesc.indexOf('M');
+
+/**
+ * Convert a moment.duration into an es
+ * compatible expression, and provide
+ * associated metadata
+ *
+ * @param  {moment.duration} duration
+ * @return {object}
+ */
+export function convertDurationToNormalizedEsInterval(duration) {
```

Comment:
comment: I'm a little curious what this is needed for.  Perhaps sent to the watcher APIs to get the visualization data?  I think to get this PR merged, this is fine, but if it IS used when getting/processing the visualization data, we may be able to avoid some of this with our replacement of the watcher API.  We'll see when we get there ...